main.go

```diff
@@ -73,13 +82,46 @@ func main() {
 		os.Exit(0)
 	}
 
+	rand.Seed(time.Now().UnixNano())
+
+	if *templateFlag != "" {
+		t, err := fakedata.ParseTemplate(*templateFlag)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		tmp = t
+	}
+
+	if isPipe() {
+		tp, err := ioutil.ReadAll(os.Stdin)
+
+		if err != nil {
+			fmt.Printf("Unable to read input: %s", err)
+			os.Exit(1)
+		}
+
+		t, err := fakedata.ParseTemplateFromPipe(string(tp))
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		tmp = t
+	}
+	// Execute a template if there is one
+	if tmp != nil {
```

Comment:
I agree that `tmp` sounds more like temporary. However, `tmpl` throws a misspell warning with the Go tools (and with my IDE). Maybe we need to go with `t`? I'm currently not sure if misspells break ci or not.